### PR TITLE
Update type declarations on ClassLoader

### LINKED ISSUE #11482: Wrong return type on Classloader::getPrefixes
I haven't PRed because I'm not sure what the type is in all cases but `string[]` appears to be incorrect, we get a multidimensional array back

https://github.com/composer/composer/blob/52f6f74b7c342f7b90be9c1a87e183092e8ab452/src/Composer/Autoload/ClassLoader.php#L116

### PR DESCRIPTION
**Pull request by Seldaek**
Fixes #11482

### CODE REVIEW

File: src/Composer/Autoload/ClassLoader.php [hunk 1]
```diff
@@ -45,66 +45,63 @@ class ClassLoader
...
 
     // PSR-0
     /**
-     * @var array[]
-     * @psalm-var array<string, array<string, string[]>>
+     * List of PSR-0 prefixes
+     *
+     * Structured as array('F (first letter)' => array('Foo\Bar (full prefix)' => array('path', 'path2')))
+     *
+     * @var array<string, array<string, string[]>>
```
Comment (stof) [inline src/Composer/Autoload/ClassLoader.php:71]: should this use `array<string, array<string, list<string>>>` to be even more precise ?

File: src/Composer/Autoload/ClassLoader.php [hunk 2]
```diff
@@ -125,44 +122,39 @@ public function getPrefixes()
     }
 
     /**
-     * @return array[]
-     * @psalm-return array<string, array<int, string>>
+     * @return array<string, array<int, string>>
```
Comment (stof) [inline src/Composer/Autoload/ClassLoader.php:125]: `array<string, list<string>>` would be more precise

Comment (Seldaek) [reply src/Composer/Autoload/ClassLoader.php:125]: Yup, thanks 45368a5bd

Comment (Seldaek) [reply src/Composer/Autoload/ClassLoader.php:125]: Sorry 3f385d466fef9bdd355a6ec01ec7c1875bd95f92

**Pull request merged**